### PR TITLE
Add the basic idea for showing the sidebar on mobile

### LINKED ISSUE #29: No Menubar on mobile
Currently there is a button to open the sidebar on mobile, but it is not linked to showing the sidebar, and the sidebar is not visible if I manually disable the `display: none;` via devtools.

### PR DESCRIPTION
**Pull request by jasikpark**
I figure it'd be a good idea to implement the CSS along these lines: https://adrianroselli.com/2021/06/using-css-to-enforce-accessibility.html, though it's a bit messy at the moment, the basic setup is done for making the sidebar navigation visible on mobile, but I haven't figured out the grid and everything.

Also topical: https://inclusive-components.design/menus-menu-buttons/#navigationmenubuttons

Fixes #29

### CODE REVIEW

File: src/components/DocSidebar.tsx [hunk 1]
```diff
@@ -40,15 +51,23 @@ const DocSidebar: FunctionalComponent<{ headers: any[]; editHref: string }> = ({
   return (
     <nav>
       <div>
-        <h2 class="heading">Contents</h2>
+        <h2 className="heading">Contents</h2>
```
Comment (natemoo-re) [inline src/components/DocSidebar.tsx:54]: Preact actually supports `class` if you prefer that!

File: src/components/MenuToggle.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      sidebar.classList.remove('hidden');
+    } else {
+      sidebar.classList.add('hidden');
+    }
+  }, [sidebarShown]);
+
+  return (
+    <button
+      type="button"
+      aria-pressed={sidebarShown ? 'true' : 'false'}
```
Comment (natemoo-re) [inline src/components/MenuToggle.tsx:20]: Love to see it! ❤️

**Review by natemoo-re — COMMENTED** [review]
Lookin' really great! Let me know if you want some guidance on how to style things?

File: src/components/DocSidebar.tsx (hunk 1 above)
Comment (jasikpark) [reply src/components/DocSidebar.tsx:54]: Oh! My editor was yelling at me, I guess it thought the file was vanilla React then.

**Comment by jasikpark** [comment]
> Lookin' really great! Let me know if you want some guidance on how to style things?

Please!

**Comment by FredKSchott** [comment]
Love the ARIA + code improvements to the toggle!

FYI when I clicked the sidebar menu on mobile it didn't do anything. I just merged a change to add support for a ".mobile-sidebar-hidden" class on the body, so you should be able to revert your styling changes to `src/layouts/Main.astro` and then just hook into that instead:

```
  useEffect(() => {
    if (sidebarShown) {
      document.body.classList.remove('mobile-sidebar-hidden');
    } else {
      document.body.classList.add('mobile-sidebar-hidden');
    }
  }, [sidebarShown]);
```

**Comment by jasikpark** [comment]
> Love the ARIA + code improvements to the toggle!
> 
> FYI when I clicked the sidebar menu on mobile it didn't do anything. I just merged a change to add support for a ".mobile-sidebar-hidden" class on the body, so you should be able to revert your styling changes to `src/layouts/Main.astro` and then just hook into that instead:
> 
> ```
>   useEffect(() => {
>     if (sidebarShown) {
>       document.body.classList.remove('mobile-sidebar-hidden');
>     } else {
>       document.body.classList.add('mobile-sidebar-hidden');
>     }
>   }, [sidebarShown]);
> ```
> 
> If you can get that working, this should be good to go!

That's what my CSS does via `.sidebar.hidden` -> `.sidebar`, the problem is that it *is* showing the sidebar, but it is `8px` wide because of the grid.

```css
.layout {
        display: grid;
        grid-auto-flow: column;
        grid-template-columns: 
          minmax(var(--gutter), 1fr)  /** where --gutter = 0.5rem or 8px */
          minmax(0, var(--max-width)) 
          minmax(var(--gutter), 1fr);
        gap: 1em;
      }
```

**Comment by FredKSchott** [comment]
I think my `mobile-sidebar-hidden` change fixed this! In the mobile view, the visible sidebar is now fixed to the full page size on mobile, and not a part of the grid anymore. Try rebasing and confirming if it is fixed

**Comment by jasikpark** [comment]
I didn't get it to work yet, but I pushed the changes that I thought were meant to get it working!

**Comment by FredKSchott** [comment]
LGTM! It's working for me on the docs page, we just have some copy-paste between the main layout and `index.astro` that we need to get rid of. I'll merge this and then make that change as a fast follow.

**Comment by FredKSchott** [comment]
Done in 78e01ac - I had to make one other change to the body to prevent page scrolling behind the sidebar (double scroll bars) when the sidebar was open that made the commit bigger than expected.

So glad that this is fixed: This was our last remaining blocker to getting the doc site in front of people!

**Pull request merged**